README
/* ######################################################################################## */
/* Mailer Group Solution -- Multi-User Mass Mailing System with opt out and Admin abilities */
/* Copyright (C) 2008  Jon Harris                                                           */
/*                                                                                          */
/* This program is free software; you can redistribute it and/or                            */
/* modify it under the terms of the GNU General Public License                              */
/* as published by the Free Software Foundation.                                            */
/*                                                                                          */
/* This program is distributed in the hope that it will be useful,                          */
/* but WITHOUT ANY WARRANTY; without even the implied warranty of                           */
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the                            */
/* GNU General Public License for more details.                                             */
/*                                                                                          */
/* You should have received a copy of the GNU General Public License                        */
/* along with this program; if not, write to the Free Software                              */
/* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.          */
/* ######################################################################################## */
REQUIREMENTS:
MySQL5(maybe 4, but I coded on 5)
PHP5(same as above)
Web Server able to serve out PHP files
basic knowledge of modifying a config file



You can contact the developer by emailing: webmaster@jonsjava.com
or visit http://jonsjava.com




INSTALLATION
The configuration is pretty straightforward:
Step 1: create a MySQL database
Step 2: modify config.inc.php
	change $host to the host of your mysql server (usually won't need to change, though)
	change $db to reflect the database you just created
	change $db_user to the database user that has rights to that database.
	change $db_password to the afore mentioned database users password
	change the $web_addr to the address of your site (not the folder, that comes later, and no trailing slash),
	change the $server_root to the path (web root) of your install. if your install is http://example.com/mailer/,
		web root is /mailer/ (I know, I said server root, but I had a typo, and didn't wanna go through
		changing all the scripts to reflect the true phrase)
	change $list_name to whatever you want to name this install. This is how the database distinguishes all the data
		from one install to the next (if you use the same database for multiple installs) Don't leave this field blank
Step 3: open the installer in a web browser (http://example.com/<installed folder>/installer)
	follow the instructions
	REMOVE THE INSTALLER FOLDER AFTER YOU ARE COMPLETE!
Step 4: Create some users, moderators, and supervisors.
Step 5: Test it out before going live




NOTES:
The system has a "shutdown system" feature. If you run that script, nobody will be able to log into the user portion,
but you will still be able to log into the admin portion. When you are installing, I reccomend you use a different system
password from your admin password, and keep it secret. 

You will need to create your own terms of service after installing, because I'm no lawyer, and I don't wanna write
like a lawyer.

Updates will come as I get the time. If you would like to contribute to this project, feel free to contact me, and I will
look at your work. This project took about a week to write, and another week to fine-tune. If you can do better, by all means,
show me!  I have a major headache after all this.

Time to get a beer.

CHANGELOG
/* ######################################################################################## */
/* Mailer Group Solution -- Multi-User Mass Mailing System with opt out and Admin abilities */
/* Copyright (C) 2008  Jon Harris                                                           */
/*                                                                                          */
/* This program is free software; you can redistribute it and/or                            */
/* modify it under the terms of the GNU General Public License                              */
/* as published by the Free Software Foundation.                                            */
/*                                                                                          */
/* This program is distributed in the hope that it will be useful,                          */
/* but WITHOUT ANY WARRANTY; without even the implied warranty of                           */
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the                            */
/* GNU General Public License for more details.                                             */
/*                                                                                          */
/* You should have received a copy of the GNU General Public License                        */
/* along with this program; if not, write to the Free Software                              */
/* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.          */
/* ######################################################################################## */

PLEASE REFER TO THE README FILE FOR INSTALL INSTRUCTIONS

CURRENT VERSION(v1.18)
-------------------------------------------------------------

-----
/root_folder/index.php
Added instructions to handle storing client names
Reworked some code so the on show_all_contacts, you will see "Contact name(Contact@email)"


----
/root_folder/add_contacts.php
Added script to handle storing client names (mainly just added an array, and
modified the query so it would store it)

Modified the mailer to add the contacts name to the verification email


----
/root_folder/send_email.php
modified the to field to add the contacts name
modified the admin to add the mail groups name


----
/root_folder/admin/add_user.php
added group_name to the user add sql


----
/root_folder/admin/update_user.php
populated file (forgot to in last release) works now!


----
/root_folder/admin/index.php
added group_name to add_user and show_all_users. 
Changed username to read-only (no need to change users names).


----
/root_folder/installer/index.php
Added the group_name row to the users table
added the updater script, for people upgrading from the previous version (v1)


----


-------------------------------------------------------------
PREVIOUS VERSION(v1)
----


First release
-------------------------------------------------------------



send_email.php
<?php
session_start();
/* ######################################################################################## */
/* Mailer Group Solution -- Multi-User Mass Mailing System with opt out and Admin abilities */
/* Copyright (C) 2008  Jon Harris                                                           */
/*                                                                                          */
/* This program is free software; you can redistribute it and/or                            */
/* modify it under the terms of the GNU General Public License                              */
/* as published by the Free Software Foundation.                                            */
/*                                                                                          */
/* This program is distributed in the hope that it will be useful,                          */
/* but WITHOUT ANY WARRANTY; without even the implied warranty of                           */
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the                            */
/* GNU General Public License for more details.                                             */
/*                                                                                          */
/* You should have received a copy of the GNU General Public License                        */
/* along with this program; if not, write to the Free Software                              */
/* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.          */
/* ######################################################################################## */
if ($_POST['security_code'] == $_SESSION['code']){
	$_SESSION['data'] = null;
if (isset($_SESSION['username'])){
	if ($_SESSION['is_valid']  == true){
include("config.inc.php");
include("MIME.class.php");
$search_sql = @mysql_query("SELECT * FROM {$_SESSION['username']}_table_{$list_name} WHERE verified =1;");
$sent_mail_count = 0;
$current_month = date("n");
$current_year = date("Y");
while($row = @mysql_fetch_assoc($search_sql)) {
	$admin = $SESSION_['group_name']."<".$_SESSION['email_address'].">";
	$to  = $row['name']."<".$row['email_address'].">";
	$subject = $_POST['subject'];
	$message = "<!DOCTYPE html PUBLIC '-//W3C//DTD XHTML 1.0 Transitional//EN'
	'http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd'>
	<html>
	<head>
	<title>".$subject."</title>
	</head>";
	$message_clean1 = str_replace("\'", "'", $_POST['message']);
        $message_clean2 = str_replace('"', "'", $message_clean1);
        $message_clean3 = str_replace("\v", "", $message_clean2);
        $message_clean4 = str_replace("\'", "'", $message_clean3);
	$message .= $message_clean4;
	$message .= "<br /><br /><br />If you wish to be removed from our mailing list, please go to: \n<a href='".$site_address."remove.php?id=".$row['id']."&gid=".$_SESSION['group_id']."'> ".$web_addr.$server_root."remove.php?id=".$row['id']."&gid=".$_SESSION['group_id']."</a> \n and you will be removed immediately.";
	$message .= "\n</body></html>";
	$mime = new MIME_mail($_SESSION['email_address'], $to, $subject);
	$mime->attach($message, "", HTML, BASE64);
	$mime->send_mail();
	++$sent_mail_count;
}
$sql = "UPDATE `system_stats_{$list_name}` SET `total_sent` = `total_sent` + {$sent_mail_count}; UPDATE `users_{$list_name}` SET `pieces_sent` = `pieces_sent` + 1 WHERE uname = '{$_SESSION['username']}'; UPDATE `users_{$list_name}` SET `total_sent` = `total_sent` + {$sent_mail_count} WHERE `uname` = '{$_SESSION['username']}';";
mysql_query($sql);
$sql = "SELECT * from `system_stats_{$list_name}`;";
$results = mysql_query($sql);
$row = mysql_fetch_assoc($results);
$system_total_sent = $row['total_sent'];
$sql = "UPDATE system SET `total_sent`= `total_sent` + {$sent_mail_count} WHERE `name`='{$list_name}';";
$sql = "SELECT * FROM users_{$list_name} WHERE uname='".$_SESSION['username']."';";
$result = mysql_query($sql);
$user_row = mysql_fetch_assoc($result);
$user_history = $user_row['logs'];
$new_logs = $user_history;
$new_logs .= "DATE: ".$ccdate."\n";
$new_logs .= "Time: ".date("H:i.s")."\n";
$new_logs .= "Sent E-Mail \n";
$new_logs .= "Subject: ".$subject."\n";
$new_logs .= "Sent to ".$sent_mail_count." Email Addresses\n";
$new_logs .= "--------------------------------------------------------\n\n\n";
$sql = "UPDATE users_{$list_name} SET logs='{$new_logs}', `total_sent` = `total_sent` + 1 WHERE uname='{$_SESSION['username']}';";
mysql_query($sql);
$sql = "SELECT * from users_{$list_name} WHERE `uname` = '{$_SESSION['username']}';";
$result = mysql_query($sql);
$row = mysql_fetch_assoc($result);
$user_total_sent = $row['total_sent'];
if ($system_total_sent < $user_total_sent){
	$sql = "UPDATE `system_stats_{$list_name}` SET `top_sender`='{$_SESSION['username']}', `top_sender_total` = '{$user_total_sent}';";
	mysql_query($sql);
}
header("location:index.php?message=sent");
exit();
}

else{
	header("location:index.php");
	exit();
}
	}
}

else{
	$message_clean1 = str_replace("\'", "'", $_POST['message']);
	$message_clean2 = str_replace('"', "'", $message_clean1);
	$message_clean3 = str_replace("\v", "", $message_clean2);
	$message_clean4 = str_replace("\'", "'", $message_clean3);
	$message .= $message_clean4;
	$_SESSION['data'] = $message;
	header("location:index.php?error=captcha");
}

?>
